src/main/java/org/folio/spring/filter/LoggingRequestFilter.java

```diff
@@ -0,0 +1,126 @@
+package org.folio.spring.filter;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.web.servlet.filter.OrderedFilter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.GenericFilterBean;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import org.folio.spring.utils.MultiReadHttpServletRequestWrapper;
+
+@Log4j2
+@Component
+@ConditionalOnProperty(
+  prefix = "folio.logging.request",
+  name = "enabled",
+  havingValue = "true"
+)
+public class LoggingRequestFilter extends GenericFilterBean implements OrderedFilter {
+
+  private static final String START_TIME_ATTR = "startTime";
+  private final Level level;
+  private int order = REQUEST_WRAPPER_FILTER_MAX_ORDER + 3;
+
+  public LoggingRequestFilter(@Value("${folio.logging.request.level: FULL}") Level level) {
```

Comment:
The default level should be basic, not FULL. Because, just for instance you don't use DEGUG or TRACE log level by default. Usually default is info.